client/src/lib/data.ts

```diff
@@ -225,14 +234,79 @@ export async function readCalendar(calendarId: string): Promise<Calendar> {
   return calendar;
 }
 
-export type Mark = {
-  markId: number;
+type CreateCalParams = {
+  type: string;
+  name: string;
+  color: string;
+  desc?: string;
+  goal: number;
+};
+export async function createCal({
```

Comment:
Why not just pass in a Calendar? There are several ways to work with this.

* You could rename CreateCalParams to be UnsavedCalendar and then change Calendar to be `UnsavedCalendar & { calendarId: number }`
* You could make calendarId in Calendar optional
* You could change CreateCalParams to be `Partial<Calendar>`